builtin/plugin/backend.go

```diff
@@ -53,16 +73,24 @@ type backend struct {
 
 	// Used to detect if we already reloaded
 	canary string
+
+	// Used to detect if plugin is set
+	loaded bool
 }
 
 func (b *backend) reloadBackend() error {
+	b.Logger().Trace("plugin: reloading plugin backend", "plugin", b.config.Config["plugin_name"])
+	return b.startBackend()
+}
+
+// startBackend starts a plugin backend
+func (b *backend) startBackend() error {
 	pluginName := b.config.Config["plugin_name"]
-	b.Logger().Trace("plugin: reloading plugin backend", "plugin", pluginName)
 
 	// Ensure proper cleanup of the backend (i.e. call client.Kill())
 	b.Backend.Cleanup()
 
-	nb, err := bplugin.NewBackend(pluginName, b.config.System, b.config.Logger)
+	nb, err := bplugin.NewBackend(pluginName, b.config.System, b.config.Logger, false)
 	if err != nil {
 		return err
 	}
```

Comment:
At this point, please check the type and special paths against the values retrieved from when it was run in metadata mode and make sure they match.

Reply:
This should be addressed. It is checked further down in the `b.loaded` conditional.